File: rxbinding/src/main/java/com/jakewharton/rxbinding2/widget/AdapterViewSelectionEvent.java
package com.jakewharton.rxbinding2.widget;

import androidx.annotation.NonNull;
import android.widget.AdapterView;

public abstract class AdapterViewSelectionEvent {
  AdapterViewSelectionEvent() {
  }

  /** The view from which this event occurred. */
  @NonNull public abstract AdapterView<?> view();
}

File: rxbinding/src/main/java/com/jakewharton/rxbinding3/widget/AdapterViewSelectionEvent.kt
package com.jakewharton.rxbinding3.widget

import android.view.View
import android.widget.AdapterView

sealed class AdapterViewSelectionEvent {
  /** The view from which this event occurred.  */
  abstract val view: AdapterView<*>
}

data class AdapterViewItemSelectionEvent(
  override val view: AdapterView<*>,
  val selectedView: View?,
  val position: Int,
  val id: Long
) : AdapterViewSelectionEvent()

data class AdapterViewNothingSelectionEvent(
  override val view: AdapterView<*>
) : AdapterViewSelectionEvent()
